src/fakefactors.cxx

```diff
@@ -600,5 +600,472 @@ fakefactor_sm_lt(ROOT::RDF::RNode df, const std::string &outputname,
+                          {qcd_outputname, wjets_outputname, ttbar_outputname});
+    return df4;
+}
+
+/// Function to calculate fake factors with correctionlib.
+/// In difference to the NMSSM version, njets is used for the
+/// fraction binning, and an additional split in deltaR is
+/// applied for wjets.
+///
+/// \param df the dataframe to add the quantity to
+/// \param outputname name of the output column for the fake factor
+/// \param qcd_outputname name of the output column for the qcd fake factor
+/// \param wjets_outputname name of the output column for the wjets fake factor
+/// \param ttbar_outputname name of the output column for the ttbar fake factor
+/// \param tau_pt pt of the hadronic tau in the tau pair
+/// \param njets number of good jets in the event
+/// \param lep_mt transverse mass of the leptonic tau in the tau pair
+/// \param lep_pt pt of the leptonic tau in the tau pair
+/// \param lep_iso isolation of the leptonic tau in the tau pair
+/// \param m_vis visible mass of the tau pair
+/// \param delta_r distance in eta-phi between the two taus
+/// \param variation name of the uncertainty variation or nominal
+/// \param ff_file correctionlib json file with the fake factors
+/// \param ff_corr_file correctionlib json file with corrections for the fake
+/// factors
+///
+/// \returns a dataframe with the fake factors
+ROOT::RDF::RNode
+fakefactor_sm_lt(ROOT::RDF::RNode df, const std::string &outputname,
+                 const std::string &qcd_outputname,
+                 const std::string &wjets_outputname,
+                 const std::string &ttbar_outputname, const std::string &tau_pt,
+                 const std::string &njets, const std::string &lep_mt,
+                 const std::string &lep_pt, const std::string &lep_iso,
+                 const std::string &m_vis, const std::string &delta_r,
+                 const std::string &variation, const std::string &ff_file,
+                 const std::string &ff_corr_file) {
+    Logger::get("SM FakeFactor")
+        ->debug("Setting up functions for fake factor evaluation with "
+                "correctionlib");
+    Logger::get("SM FakeFactor")->debug("Variation - Name {}", variation);
+    // load the corrections from the correctionlib json file
+    auto qcd =
+        correction::CorrectionSet::from_file(ff_file)->at("QCD_fake_factors");
+    auto wjets =
+        correction::CorrectionSet::from_file(ff_file)->at("Wjets_fake_factors");
+    auto ttbar =
+        correction::CorrectionSet::from_file(ff_file)->at("ttbar_fake_factors");
+    auto fractions =
+        correction::CorrectionSet::from_file(ff_file)->at("process_fractions");
+
+    auto qcd_lep_pt_closure = correction::CorrectionSet::from_file(ff_corr_file)
+                                  ->at("QCD_non_closure_lep_pt_correction");
+    auto qcd_lep_iso_closure =
+        correction::CorrectionSet::from_file(ff_corr_file)
+            ->at("QCD_non_closure_lep_iso_correction");
```

Comment:
Since I renamed some of the corrections in the FF framework (latest commit) this might not be correct anymore. (this also applies for other corrections).

Reply:
Can you point me to the naming changes ? :)

Reply:
I think you can look at this dict here https://github.com/nshadskiy/TauFakeFactors/blob/main/helper/correctionlib_json.py#L8-L25
I basically added `leading` or `subleading` to the naming corresponding to first and second lepton (e, mu, tau). I think I did it like this because it was easier to be consistent in the tt channel.

Reply:
To be save you can look at the summary of the correctionlib files you are using. It just has to be consistent with the naming there.